gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -135,33 +160,74 @@ impl GatewayLdkClient {
             esplora_client: esplora_client::Builder::new(esplora_server_url).build_async()?,
             event_handler_task_handle,
             htlc_stream_receiver_or: Some(htlc_stream_receiver),
+            payment_subscriptions,
         })
     }
 
     async fn handle_next_event(
         node: &ldk_node::Node,
         htlc_stream_sender: &Sender<InterceptPaymentRequest>,
+        payment_subscriptions: &Arc<
+            Mutex<
+                HashMap<
+                    bitcoin30::hashes::sha256::Hash,
+                    tokio::sync::oneshot::Sender<PaymentOutcome>,
+                >,
+            >,
+        >,
```

Comment:
This is ugly...can we make this a defined type?

Reply:
Done